File: tests/aster/x/kotlin/cross_join_filter.kt
fun main() {
    val nums = mutableListOf(1, 2, 3)
    val letters = mutableListOf("A", "B")
    val pairs = run()
    println("--- Even pairs ---")
    for (p in pairs) {
        println(listOf(p["n"], p["l"]).joinToString(" "))
    }
}

File: tests/aster/x/php/cross_join_filter.php
<?php
$nums = [1, 2, 3];
$letters = ["A", "B"];
;
foreach ($nums as $n) {
    foreach ($letters as $l) {
        if ($n % 2 == 0) {
             = ["n" => $n, "l" => $l];
        }
    }
}
echo "--- Even pairs ---", PHP_EOL;
foreach ($pairs as $p) {
    echo  . (is_float($p["l"]) ? json_encode($p["l"], 1344) : $p["l"]), PHP_EOL;
}
